localstack/services/firehose/provider.py

```diff
@@ -317,7 +322,12 @@ def _startup():
                         wait_until_started=True,
                         ddb_lease_table_suffix="-firehose",
                     )
-                    store.kinesis_listeners[delivery_stream_name] = process
+
+                    account_region_tuple: tuple[str, str] = context.account_id, context.region
+                    listener_backend: dict[
+                        str, KinesisProcessorThread
+                    ] = self.kinesis_listeners.setdefault(account_region_tuple, {})
+                    listener_backend[delivery_stream_name] = process
```

Comment:
nit: How do you feel about simplifying the type of `kinesis_listeners` a bit to `dict[str, KinesisProcessorThread]`? Then we could simply use the `kinesis_stream_arn` as the index

Reply:
As per the comment above, I wanted to preserve a sort of multi-account behavior.

Reply:
Ok, makes sense. Using the stream ARN as key could achieve the same effect (as the ARN encodes account ID + region), but again, not critical - feel free to merge as-is as well!

Reply:
Good point. We can totally simplify then the `kinesis_listeners` signature and get rid of the tuple. That's a sweet simplification 👍